From 93f695bfe52dd725841fcca71ed4e14e9a90e1dc Mon Sep 17 00:00:00 2001
From: Abinaya Kalaiselvan <akalaise@codeaurora.org>
Date: Wed, 25 Nov 2020 19:58:58 +0530
Subject: [PATCH] hostapd: fix int-in-bool-context Werror

Add changes to fix the below compilation error.

./src/rsn_supp/wpa.c:621:35: error: ?: using integer constants
in boolean context, the expression will always evaluate
to 'true' [-Werror=int-in-bool-context]

Signed-off-by: Abinaya Kalaiselvan <akalaise@codeaurora.org>
---
 src/rsn_supp/wpa.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/rsn_supp/wpa.c b/src/rsn_supp/wpa.c
index 3205527..88c1e3e 100644
--- a/src/rsn_supp/wpa.c
+++ b/src/rsn_supp/wpa.c
@@ -615,10 +615,11 @@ static int wpa_handle_ext_key_id(struct wpa_sm *sm,
 {
 	if (sm->ext_key_id) {
 		u16 key_id;
+		u8 use_ext_key_id;
 
 		if (!kde->key_id) {
-			wpa_msg(sm->ctx->msg_ctx,
-				sm->use_ext_key_id ? MSG_INFO : MSG_DEBUG,
+			use_ext_key_id = sm->use_ext_key_id ? MSG_INFO : MSG_DEBUG;
+			wpa_msg(sm->ctx->msg_ctx, use_ext_key_id,
 				"RSN: No Key ID in Extended Key ID handshake");
 			sm->keyidx_active = 0;
 			return sm->use_ext_key_id ? -1 : 0;
-- 
2.7.4

